src/commands/deploy/verifyAppSettings.ts

```diff
@@ -71,29 +71,44 @@ export async function verifyAndUpdateAppConnectionStrings(context: IActionContex
     let didUpdate: boolean = false;
     switch (durableStorageType) {
         case DurableBackend.Netherite:
-            const updatedNetheriteConnection: boolean = updateConnectionStringIfNeeded(context, remoteProperties, ConnectionKey.EventHubs, context[ConnectionKey.EventHubs]);
-            didUpdate ||= updatedNetheriteConnection;
+            if (context.newEventHubsNamespaceConnectionSettingKey && context.newEventHubConnectionSettingValue) {
+                const updatedNamespaceConnection: boolean = updateConnectionStringIfNeeded(context, remoteProperties, context.newEventHubsNamespaceConnectionSettingKey, context.newEventHubsNamespaceConnectionSettingValue);
+                didUpdate ||= updatedNamespaceConnection;
+            }
+            if (context.newEventHubConnectionSettingKey && context.newEventHubConnectionSettingValue) {
+                const updatedEventHubConnection: boolean = updateConnectionStringIfNeeded(context, remoteProperties, context.newEventHubConnectionSettingKey, context.newEventHubConnectionSettingValue);
+                didUpdate ||= updatedEventHubConnection;
+            }
             break;
         case DurableBackend.DTS:
-            const updatedDTSConnection: boolean = updateConnectionStringIfNeeded(context, remoteProperties, ConnectionKey.DTS, context[ConnectionKey.DTS]);
-            const updatedDTSHub: boolean = updateConnectionStringIfNeeded(context, remoteProperties, ConnectionKey.DTSHub, context[ConnectionKey.DTSHub]);
-            didUpdate ||= updatedDTSConnection || updatedDTSHub;
+            if (context.newDTSConnectionSettingKey && context.newDTSConnectionSettingValue) {
+                const updatedDTSConnection: boolean = updateConnectionStringIfNeeded(context, remoteProperties, context.newDTSConnectionSettingKey, context.newDTSConnectionSettingValue);
+                didUpdate ||= updatedDTSConnection;
+            }
+            if (context.newDTSHubConnectionSettingKey && context.newDTSConnectionSettingValue) {
+                const updatedDTSHub: boolean = updateConnectionStringIfNeeded(context, remoteProperties, context.newDTSHubConnectionSettingKey, context.newDTSHubConnectionSettingValue);
```

Comment:
Looking at line 89 in the DTS case, you're checking if `newDTSHubConnectionSettingKey` exists but then using `newDTSConnectionSettingValue` instead of `newDTSHubConnectionSettingValue`:

```typescript
if (context.newDTSHubConnectionSettingKey && context.newDTSConnectionSettingValue) {
    const updatedDTSHub: boolean = updateConnectionStringIfNeeded(context, remoteProperties, context.newDTSHubConnectionSettingKey, context.newDTSHubConnectionSettingValue);
```

The condition checks for `newDTSHubConnectionSettingKey` but uses `newDTSConnectionSettingValue` - should be `newDTSHubConnectionSettingValue` to match the key.

Reply:
good catch!